src/mo_types/type.ml

```diff
@@ -41,6 +41,21 @@ type prim =
   | Principal
   | Region
 
+module Region_ord = struct
+  type t = Source.region
+
+  let compare l r =
+    let open Source in
+    let join_compare l r = if l = 0 then r else l in
+    let compare_pos l r =
+      join_compare
+        (compare l.file r.file)
+        (join_compare (compare l.line r.line) (compare l.column r.column))
+    in
+    join_compare (compare_pos l.left r.left) (compare_pos l.right r.right)
```

Comment:
Use nested if or match to avoid doing redundant comparisons.

Reply:
Done. ([Link to changes](https://github.com/serokell/motoko/pull/57))